Fix gcc warnings

PhenomRetroShare opened the pull request.

PhenomRetroShare committed:
Fix Gcc Warning in TransfersHandler.cpp

/libresapi/src/api/TransfersHandler.cpp:29: warning: this ‘if’ clause
does not guard... [-Wmisleading-indentation]
     if(time(0) > (mLastUpdateTS + UPDATE_PERIOD_SECONDS))

PhenomRetroShare committed:
Fix Gcc Warning in pgpkeyutil.cc

/libretroshare/src/pgp/pgpkeyutil.cc:189: warning: this ‘if’ clause does
not guard... [-Wmisleading-indentation]
     if(signature_type != 4)

PhenomRetroShare committed:
Fix Gcc Warning in rsdir.cc

/libretroshare/src/util/rsdir.cc:111: warning: comparison between signed
and unsigned integer expressions [-Wsign-compare]

PhenomRetroShare committed:
Fix Gcc Warning in ChatHandler.cpp

/libresapi/src/api/ChatHandler.cpp:1237: warning: this ‘if’ clause does
not guard... [-Wmisleading-indentation]

PhenomRetroShare committed:
Fix Gcc Warning in MainWindow.cpp

/retroshare-gui/src/gui/MainWindow.cpp:730: warning: this ‘if’ clause
does not guard... [-Wmisleading-indentation]
     if (ratesstatus)
     ^~

PhenomRetroShare committed:
Fix Gcc Warning in TransfersDialog.cpp

/retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp:126: warning:
unused parameter ‘parent’ [-Wunused-parameter]
  int columnCount(const QModelIndex &parent = QModelIndex()) const
/retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp:157: warning:
comparison between signed and unsigned integer expressions [-Wsign-
compare]
    if(row >= mDownloads.size() || !
convertTabEntryToRefPointer(row,-1,subref))
/retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp:197: warning:
unused parameter ‘orientation’ [-Wunused-parameter]
  QVariant headerData(int section, Qt::Orientation orientation, int role
= Qt::DisplayRole) const
/retroshare-gui/src/gui/FileTransfer/TransfersDialog.cpp:226: warning:
unused variable ‘coln’ [-Wunused-variable]
   int coln = index.column() ;

PhenomRetroShare committed:
Fix Gcc Warning in torstatus.cpp

/retroshare-gui/src/gui/statusbar/torstatus.cpp:65: warning: variable
‘online’ set but not used [-Wunused-but-set-variable]
  bool online ;

PhenomRetroShare committed:
Fix Gcc Warning in FileSharingHandler.cpp

/libresapi/src/api/FileSharingHandler.cpp:87: warning: dereferencing
type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  resp.mDataStream << makeKeyValue("parent_reference",
*reinterpret_cast<int*>(&dirDetails.ref));
/libresapi/src/api/FileSharingHandler.cpp:133: warning: dereferencing
type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
            << makeKeyValue("parent_reference",
*reinterpret_cast<int*>(&dirDetails.parent))
/libresapi/src/api/FileSharingHandler.cpp:134: warning: dereferencing
type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
            << makeKeyValue("reference",
*reinterpret_cast<int*>(&dirDetails.ref))
/libresapi/src/api/FileSharingHandler.cpp:277: warning: dereferencing
type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  resp.mDataStream << makeKeyValue("parent_reference",
*reinterpret_cast<int*>(&dirDetails.ref));
/libresapi/src/api/FileSharingHandler.cpp:320: warning: dereferencing
type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
            << makeKeyValue("parent_reference",
*reinterpret_cast<int*>(&dirDetails.parent))
/libresapi/src/api/FileSharingHandler.cpp:321: warning: dereferencing
type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
            << makeKeyValue("reference",
*reinterpret_cast<int*>(&dirDetails.ref))
/libresapi/src/api/FileSharingHandler.cpp:386: warning: dereferencing
type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  resp.mDataStream << makeKeyValue("parent_reference",
*reinterpret_cast<int*>(&dirDetails.ref));
/libresapi/src/api/FileSharingHandler.cpp:430: warning: dereferencing
type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
            << makeKeyValue("parent_reference",
*reinterpret_cast<int*>(&dirDetails.parent))
/libresapi/src/api/FileSharingHandler.cpp:431: warning: dereferencing
type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
            << makeKeyValue("reference",
*reinterpret_cast<int*>(&dirDetails.ref))

PhenomRetroShare committed:
Fix Gcc Warning in authssl.cc

/libretroshare/src/pqi/authssl.cc:160: warning: ‘void
dyn_destroy_function(CRYPTO_dynlock_value*, const char*, int)’ defined
but not used [-Wunused-function]
 static void dyn_destroy_function(struct CRYPTO_dynlock_value *l, const
char */*file*/, int /*line*/)
/libretroshare/src/pqi/authssl.cc:143: warning: ‘void
dyn_lock_function(int, CRYPTO_dynlock_value*, const char*, int)’ defined
but not used [-Wunused-function]
 static void dyn_lock_function(int mode, struct CRYPTO_dynlock_value *l,
const char */*file*/, int /*line*/)
/libretroshare/src/pqi/authssl.cc:121: warning: ‘CRYPTO_dynlock_value*
dyn_create_function(const char*, int)’ defined but not used [-Wunused-
function]
 static struct CRYPTO_dynlock_value *dyn_create_function(const char */
*file*/, int /*line*/)
/libretroshare/src/pqi/authssl.cc:106: warning: ‘long unsigned int
id_function()’ defined but not used [-Wunused-function]
 static unsigned long id_function(void)
/libretroshare/src/pqi/authssl.cc:92: warning: ‘void
locking_function(int, int, const char*, int)’ defined but not used [-
Wunused-function]
 static void locking_function(int mode, int n, const char */*file*/, int
/*line*/)

PhenomRetroShare committed:
Fix Gcc Warning hashstream.h

/libretroshare/src/crypto/hashstream.cc:38: warning: friend declaration
‘template<class T> librs::crypto::HashStream&
librs::crypto::operator<<(librs::crypto::HashStream&, const T&)’ is not
visible to explicit specialization
  HashStream& operator<<(HashStream& u,const std::string& s)
/libretroshare/src/crypto/hashstream.cc:44: warning: friend declaration
‘template<class T> librs::crypto::HashStream&
librs::crypto::operator<<(librs::crypto::HashStream&, const T&)’ is not
visible to explicit specialization
  HashStream& operator<<(HashStream& u,const uint64_t& n)
/libretroshare/src/crypto/hashstream.cc:58: warning: friend declaration
‘template<class T> librs::crypto::HashStream&
librs::crypto::operator<<(librs::crypto::HashStream&, const T&)’ is not
visible to explicit specialization
  HashStream& operator<<(HashStream& u,const uint32_t& n)
/libretroshare/src/crypto/hashstream.cc:72: warning: friend declaration
‘template<class T> librs::crypto::HashStream&
librs::crypto::operator<<(librs::crypto::HashStream&, const T&)’ is not
visible to explicit specialization
  HashStream& operator<<(HashStream& u,const uint8_t& n)

PhenomRetroShare committed:
Fix Gcc Warning in rsturtleitem_test.cc

/tests/unittests/libretroshare/serialiser/rsturtleitem_test.cc:45:
warning: no return statement in function returning non-void [-Wreturn-
type]
 }

csoler commented on libretroshare/src/util/rsdir.cc:
```diff
@@ -24,31 +24,31 @@
-#include "retroshare/rstypes.h"
-#include "retroshare/rsnotify.h"
+#include "folderiterator.h"
+#include "rsdir.h"
```
why remove the util/ ? it makes the include line more confusing.

PhenomRetroShare replied:
Done

csoler commented on libresapi/src/api/FileSharingHandler.cpp:
```diff
@@ -84,7 +84,9 @@ void FileSharingHandler::handleGetSharedDir(Request& /*req*/, Response& resp)
 	mRsFiles->RequestDirDetails(NULL, dirDetails, RS_FILE_HINTS_LOCAL);
 
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
 	resp.mDataStream << makeKeyValue("parent_reference", *reinterpret_cast<int*>(&dirDetails.ref));
```
The proper way to fix this is not to hide the warning, but to cast using intptr_t:

resp.mDataStream << makeKeyValue("...", (int*)(intptr_t)(dirDetails.ref));

PhenomRetroShare replied:
Done but use:
*reinterpret_cast<int*>(reinterpret_cast<intptr_t>(&dirDetails.ref))
Yours failed to compiles.

csoler commented on libretroshare/src/gxstrans/p3gxstransitems.cc:
```diff
@@ -48,7 +48,9 @@ void OutgoingRecord_deprecated::serial_process(RsGenericSerializer::SerializeJob
 	RS_REGISTER_SERIAL_MEMBER(mailItem);
 	RS_REGISTER_SERIAL_MEMBER(mailData);
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
 	RS_REGISTER_SERIAL_MEMBER_TYPED(clientService, uint16_t);
```
This warning tells something concerning. We should not hide it, but rather fix the wrong cast below.

PhenomRetroShare replied:
Done, only remove them, now no warnings???

G10h4ck replied:
I would not concern much about those casts as PR #1155 introduce a new system that doesn't suffer of that kind of problems

PhenomRetroShare replied:
Do you says that all of these code could be removed?

G10h4ck replied:
no absolutely!! I say that those warning disappears when https://github.com/RetroShare/RetroShare/pull/1155 will be merged as i have found a better way to handle enums in serialization there

PhenomRetroShare committed:
Fix Gcc Warning in ftchunkmap.cc

/libretroshare/src/ft/ftchunkmap.cc:260: warning: unused parameter
‘size_hint’ [-Wunused-parameter]
 bool ChunkMap::reAskPendingChunk(const RsPeerId& peer_id,uint32_t
size_hint,uint64_t& offset,uint32_t& size)

PhenomRetroShare committed:
Fix Gcc Warning in RsCollectionDialog.cpp

/home/phenom/GIT/RetroShare/trunk/retroshare-gui/src/gui/common/
RsCollectionDialog.cpp:645: warning: comparison between signed and
unsigned integer expressions [-Wsign-compare]
  for(uint32_t i=0;i<ui._fileEntriesTW->topLevelItemCount();++i)

PhenomRetroShare committed:
Fix Clang Warning in rsexpr.h

/libretroshare/src/retroshare/rsexpr.h:188: warning:
'RsRegularExpression::StringExpression::toStdString' hides overloaded
virtual function [-Woverloaded-virtual]
        virtual std::string toStdString(const std::string& varstr)
const;
/libretroshare/src/retroshare/rsexpr.h:204: warning:
'RsRegularExpression::RelExpression<int>::toStdString' hides overloaded
virtual function [-Woverloaded-virtual]
        virtual std::string toStdString(const std::string& typestr)
const;
/libretroshare/src/retroshare/rsexpr.h:282: warning:
'RsRegularExpression::NameExpression::toStdString' hides overloaded
virtual function [-Woverloaded-virtual]
        virtual std::string toStdString() const { return
StringExpression::toStdString("NAME"); }
/libretroshare/src/retroshare/rsexpr.h:297: warning:
'RsRegularExpression::PathExpression::toStdString' hides overloaded
virtual function [-Woverloaded-virtual]
        virtual std::string toStdString()const { return
StringExpression::toStdString("PATH"); }
/libretroshare/src/retroshare/rsexpr.h:312: warning:
'RsRegularExpression::ExtExpression::toStdString' hides overloaded
virtual function [-Woverloaded-virtual]
        virtual std::string toStdString()const { return
StringExpression::toStdString("EXTENSION"); }
/libretroshare/src/retroshare/rsexpr.h:327: warning:
'RsRegularExpression::HashExpression::toStdString' hides overloaded
virtual function [-Woverloaded-virtual]
        virtual std::string toStdString() const { return
StringExpression::toStdString("HASH"); }
/libretroshare/src/retroshare/rsexpr.h:349: warning:
'RsRegularExpression::DateExpression::toStdString' hides overloaded
virtual function [-Woverloaded-virtual]
        virtual std::string toStdString() const { return
RelExpression<int>::toStdString("DATE"); }
/libretroshare/src/retroshare/rsexpr.h:366: warning:
'RsRegularExpression::SizeExpression::toStdString' hides overloaded
virtual function [-Woverloaded-virtual]
        virtual std::string toStdString() const { return
RelExpression<int>::toStdString("SIZE"); }
/libretroshare/src/retroshare/rsexpr.h:383: warning:
'RsRegularExpression::SizeExpressionMB::toStdString' hides overloaded
virtual function [-Woverloaded-virtual]
        virtual std::string toStdString() const { return
RelExpression<int>::toStdString("SIZE"); }
/libretroshare/src/retroshare/rsexpr.h:399: warning:
'RsRegularExpression::PopExpression::toStdString' hides overloaded
virtual function [-Woverloaded-virtual]
        virtual std::string toStdString() const { return
RelExpression<int>::toStdString("POPULARITY"); }

PhenomRetroShare committed:
Fix Clang Warning in authssl.cc

/libretroshare/src/pqi/authssl.cc:66: warning: unused variable
'RS_SSL_HANDSHAKE_DIAGNOSTIC_KEY_ALGORITHM_NOT_ACCEPTED' [-Wunused-
const-variable]
const uint32_t RS_SSL_HANDSHAKE_DIAGNOSTIC_KEY_ALGORITHM_NOT_ACCEPTED  =
0x08 ;
               ^

PhenomRetroShare committed:
Fix Clang Warning in FriendSelectionWidget.cpp

/retroshare-gui/src/gui/common/FriendSelectionWidget.cpp:250: warning:
unused variable 'vit' [-Wunused-variable]
        std::vector<RsGxsIdGroup>::iterator vit;

PhenomRetroShare committed:
Fix Clang Warning in ServerPage.cpp

/retroshare-gui/src/gui/settings/ServerPage.cpp:74: warning: unused
variable 'TAB_HIDDEN_SERVICE_OUTGOING' [-Wunused-const-variable]
const static uint32_t TAB_HIDDEN_SERVICE_OUTGOING = 0;
/retroshare-gui/src/gui/settings/ServerPage.cpp:77: warning: unused
variable 'TAB_NETWORK' [-Wunused-const-variable]
const static uint32_t TAB_NETWORK                 = 0;

PhenomRetroShare committed:
Fix Clang Warning in ConnectProgressDialog.cpp

/retroshare-gui/src/gui/connect/ConnectProgressDialog.cpp:94: warning:
unused variable 'CONNECT_STATE_INIT' [-Wunused-const-variable]
const uint32_t CONNECT_STATE_INIT               = 0;
/retroshare-gui/src/gui/connect/ConnectProgressDialog.cpp:109: warning:
unused variable 'CONNECT_LOOKUP_SEARCH' [-Wunused-const-variable]
const uint32_t CONNECT_LOOKUP_SEARCH            = 1;
/retroshare-gui/src/gui/connect/ConnectProgressDialog.cpp:135: warning:
unused variable 'CONNECT_LOOKUP_TYPICAL' [-Wunused-const-variable]
const int CONNECT_LOOKUP_TYPICAL = 200;
/retroshare-gui/src/gui/connect/ConnectProgressDialog.cpp:136: warning:
unused variable 'CONNECT_LOOKUP_SLOW' [-Wunused-const-variable]
const int CONNECT_LOOKUP_SLOW    = 300;
/retroshare-gui/src/gui/connect/ConnectProgressDialog.cpp:137: warning:
unused variable 'CONNECT_LOOKUP_PERIOD' [-Wunused-const-variable]
const int CONNECT_LOOKUP_PERIOD  = 600;

PhenomRetroShare committed:
Fix Gcc Warning in RemoteDirModel.cpp

/retroshare-gui/src/gui/RemoteDirModel.cpp:377: warning: unused
parameter ‘coln’ [-Wunused-parameter]
 QVariant RetroshareDirModel::filterRole(const DirDetails& details,int
coln) const

PhenomRetroShare committed:
Fix Gcc Warning in SharedFilesDialog.cpp

/retroshare-gui/src/gui/FileTransfer/SharedFilesDialog.cpp:1542:
warning: variable ‘flags’ set but not used [-Wunused-but-set-variable]
  FileSearchFlags flags = isRemote()?
RS_FILE_HINTS_REMOTE:RS_FILE_HINTS_LOCAL;

PhenomRetroShare committed:
Fix Clang Warning in rsplugin.h

/libretroshare/src/retroshare/rsplugin.h:72: warning: class 'RsIdentity'
was previously declared as a struct [-Wmismatched-tags]
class RsIdentity;

PhenomRetroShare committed:
Fix Clang Warning in ChatHandler.h

/libresapi/src/api/ChatHandler.h:9: warning: class 'RsIdentity' was
previously declared as a struct [-Wmismatched-tags]
class RsIdentity;

PhenomRetroShare committed:
Fix Clang Warning in MessageComposer.h

/retroshare-gui/src/gui/msgs/MessageComposer.h:34: warning: class
'RsIdentityDetails' was previously declared as a struct [-Wmismatched-
tags]
class RsIdentityDetails;

PhenomRetroShare committed:
Fix Clang Warning in IdentityHandler.h

/libresapi/src/api/IdentityHandler.h:28: warning: class 'RsIdentity' was
previously declared as a struct [-Wmismatched-tags]
class RsIdentity;

PhenomRetroShare committed:
Fix Clang Warning in rsidentity.h

/usr/include/c++/7.3.1/bits/stl_construct.h:98: warning: destructor
called on non-final 'RsGxsIdGroup' that has virtual functions but non-
virtual destructor [-Wdelete-non-virtual-dtor]
    { __pointer->~_Tp(); }

PhenomRetroShare committed:
Fix Clang Warning in rsgxsifacetypes.h

/usr/include/c++/7.3.1/ext/new_allocator.h:140: warning: destructor
called on non-final 'RsGroupMetaData' that has virtual functions but
non-virtual destructor [-Wdelete-non-virtual-dtor]
        destroy(_Up* __p) { __p->~_Up(); }

G10h4ck commented:
This PR has conflicts and stagnated, a discussion on how to make easier to review and merge warning fixing PRs has happened here https://github.com/RetroShare/RetroShare/pull/1382
